CMakeLists.txt

```diff
@@ -267,9 +271,13 @@ set(CMAKE_C_FLAGS_DEBUGOPT "")
 
 target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
         -Wuninitialized -Wshadow -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
-        -Wno-missing-braces -Wsign-compare -Wno-strict-prototypes -Wa,--noexecstack
+        -Wno-missing-braces -Wno-strict-prototypes -Wa,--noexecstack
 )
 
+if (EXPERIMENTAL_TREAT_WARNINGS_AS_ERRORS)
+    target_compile_options(${PROJECT_NAME} PRIVATE -Wsign-compare -DEXPERIMENTAL_TREAT_WARNINGS_AS_ERRORS)
```

Comment:
If EXPERIMENTAL_TREAT_WARNINGS_AS_ERRORS is already set, does it need to be passed through as a compiler flag?